app/app/src/main/kotlin/com/hedvig/app/feature/genericauth/SsnInputScreen.kt

```diff
@@ -0,0 +1,176 @@
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.safeDrawingPadding
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.text.KeyboardActions
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Surface
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.unit.dp
+import com.hedvig.android.core.designsystem.component.button.HedvigContainedButton
+import com.hedvig.android.core.designsystem.component.textfield.HedvigTextField
+import com.hedvig.android.core.designsystem.preview.HedvigPreview
+import com.hedvig.android.core.designsystem.theme.HedvigTheme
+import com.hedvig.android.core.ui.appbar.TopAppBarWithBack
+import com.hedvig.android.market.Market
+import hedvig.resources.R
+
+@Composable
+fun SSNInputScreen(
+  market: Market,
+  onUpClick: () -> Unit,
+  onInputChanged: (String) -> Unit,
+  onSubmitEmail: () -> Unit,
+  emailInput: String,
+  error: String?,
+  loading: Boolean,
+) {
+  Column(
+    modifier = Modifier
+      .safeDrawingPadding()
+      .fillMaxSize(),
+  ) {
+    TopAppBarWithBack(
+      onClick = onUpClick,
+      title = stringResource(R.string.zignsec_login_screen_title),
+    )
+    Column(
+      modifier = Modifier
+        .fillMaxSize()
+        .padding(horizontal = 16.dp)
+        .verticalScroll(rememberScrollState()),
+    ) {
+      Spacer(Modifier.height(60.dp))
+      Text(
+        text = stringResource(R.string.zignsec_login_screen_title),
+        style = MaterialTheme.typography.headlineMedium,
+      )
+      Spacer(Modifier.height(20.dp))
+      SSNTextField(
+        input = emailInput,
+        market = market,
+        onInputChanged = onInputChanged,
+        onSubmit = onSubmitEmail,
+        error = error,
+      )
+      Spacer(Modifier.weight(1f))
+      Spacer(Modifier.height(16.dp))
+      HedvigContainedButton(
+        text = stringResource(hedvig.resources.R.string.login_continue_button),
+        onClick = onSubmitEmail,
+        isLoading = loading,
+      )
+      Spacer(Modifier.height(16.dp))
+    }
+  }
+}
+
+@Composable
+private fun SSNTextField(
+  input: String,
+  market: Market,
+  onInputChanged: (String) -> Unit,
+  onSubmit: () -> Unit,
+  error: String?,
+) {
+  HedvigTextField(
+    value = input,
+    onValueChange = { newInput ->
+      val maxLengthAllowed = when (market) {
+        Market.NO -> 11
+        Market.DK -> 10
+        Market.SE -> 10
```

Comment:
Do we not tend to use the full 12 character SSN lately for such places? Would this work with 12 instead?

Oh also actually we should never end up in this screen with market being SE right? Because we default to another screen instead? If that's the case, we should just `error()` here and let it crash if it's the case, so we actually know that we got an important bug to fix.

Reply:
Thats a good point. Will change